xds/src/main/java/io/grpc/xds/GrpcXdsTransportFactory.java

```diff
@@ -84,6 +85,12 @@ public void shutdown() {
       channel.shutdown();
     }
 
+    @Override
+    public boolean isConnected() {
+      ConnectivityState state = channel.getState(false);
```

Comment:
This is resolved because you deleted this method, but I think you missed my point. `channel.getState()` can change at any time. So if you check this value three times, all within the synchronization context, it could change each time.

(And you probably meant `volatile`, not `transient`. That would avoid a data race (which wasn't a problem with channel.getState), but it wouldn't alleviate my concern about the value changing randomly while doing processing.)

Reply:
Since the variable is only set from within synchronization contexts, it can't change while doing processing (unlike the channel state) so I think we're good now that my terminology has been straightened out.